src/include/units/quantity_kind.h

```diff
@@ -268,33 +290,47 @@ template<QuantityKind QK1, QuantityKindEquivalentTo<QK1> QK2>
 }
 
 template<QuantityKind QK, Quantity Q>
-[[nodiscard]] constexpr QuantityKind auto operator*(const QK& lhs, const Q& rhs)
+[[nodiscard]] constexpr auto operator*(const QK& lhs, const Q& rhs)
```

Comment:
Why did we get rid of concepts usage here? Pure `auto` as a return type is not the best for interface documentation needs.

Reply:
A missed part of a reverted change. I'll fix that.